### ISSUE TITLE
How to do spatial interpolation ?

### ISSUE THREAD

**Issue by yjhong89** [konstantinklemmer/pe-gnn#6]
Hi thanks for sharing great works!

PE-GCN paper shows performance on 1) Spatial interpolation, 2) Spatial regression and spatial interpolation predicts outcomes from spatial coordinate alone (without any node feature)
- I wonder how I can do spatial interpolation on this repo.
- It seems that `x (node feature)` assumed to be always exist.
  - https://github.com/konstantinklemmer/pe-gnn/blob/58f43fffb731f461132ec21e9e889aa2eb746bfa/src/model.py#L402

Thanks!

**Comment by konstantinklemmer** [comment]
Hi @yjhong89! You can simply insert a dummy node feature, e.g. `x = torch.ones(n_nodes, 1)` where `n_nodes` is the number of nodes in your graph. Hope this helps!

**Comment by yjhong89** [comment]
@konstantinklemmer ,
Thanks for reply!
- 2 more questions..

1. If node features (`x`) have shape of [# nodes, d], then isn't dummy node also have to be [# nodes, d] ??
2. Would value of dummy node feature be matter? I thought value of dummy node should be zero so only coordinate PE embedding concerns forward pass. What do you think??

**Comment by konstantinklemmer** [comment]
1. Sure, but you said you dont have any node features, no? You just want to predict [lon,lat] -> y, right?
2. As long as they're all the same it doesn't matter I would assume.

**Comment by yjhong89** [comment]
Thanks for quick reply.

- I am gonna predict oil production based on PE-GCN. The thing is that I want to predict oil production where oil well not exists based on only latitude / longitude. 
- So I have node features in training and don't have node features in inference, I just want to check how spatial interpolation works in my case.

**Comment by konstantinklemmer** [comment]
Oh I see! In that case its a bit risky to have node features in train but not the test set and still train a model with the node features. In your scenario, I would run two models and compare them:
1. A pure spatial interpolation model where both x_train and x_test are torch.ones(n_nodes_{train/test}, 1).
2. One model like you suggest where you use the train node features you have. In that case you are right, you have to set your x_train to torch.ones(n_nodes,d).
Hope this helps!

**Comment by yjhong89** [comment]
Thanks for clear!

- Your second model seems to be proper for me. 
  - Train with node features, and do spatial interpolation (inference) with `x_test==torch.ones(n_nodes, d)`. Is that what you said right ??

**Comment by konstantinklemmer** [comment]
Yup that's correct! But again, I would test both approaches and compare them.

**Comment by yjhong89** [comment]
Thanks a lot!!

**Comment by yjhong89** [comment]
@konstantinklemmer , One more question.

- When we do spatial interpolation, we need to predict for unseen coordinates during training. 
- Then, is it necessary to construct adjacent matrix for unseen coordinate against whole training dataset ??
- Or is it enough using k-nearest neighbor of unseen coordinate ?? 
- If relevant code is in this repo, it would be appreciated to be shared.

Thanks!

**Comment by konstantinklemmer** [comment]
Yes so that is an interesting question. In our paper we only look at interpolation, i.e. assuming that train and test set are not geographically separate. So in our case during test time, we construct the test graph using ALL points, train and test, but then provide evaluation metrics just for test nodes. Hope this helps.

**Comment by yjhong89** [comment]
Thanks!
